src/codegen/generators/expression_generator.rs

```diff
@@ -496,27 +496,24 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
             .find_implementation(self.index)
             .ok_or_else(|| Diagnostic::cannot_generate_call_statement(operator))?;
 
+        let parameters_list = parameters
+            .as_ref()
+            .map(ast::flatten_expression_list)
+            .unwrap_or_default();
+
         //If the function is builtin, generate a basic value enum for it
         if let Some(builtin) = self
             .index
             .get_builtin_function(implementation.get_call_name())
         {
             //adr, ref, etc.
-            return builtin.codegen(
-                self,
-                parameters
-                    .as_ref()
-                    .map(ast::flatten_expression_list)
-                    .unwrap_or_default()
-                    .as_slice(),
-                operator.get_location(),
-            );
+            return builtin.codegen(self, parameters_list.as_slice(), operator.get_location());
         }
 
         let function_name = implementation.get_call_name();
         let arguments_list = self.generate_pou_call_arguments_list(
             pou,
-            parameters,
```

Comment:
I think you can avoid cloning this by simply expecting a &[AstStatement] in the methods below.